File: algorithm/boj/_1000/1011.c
#include <stdio.h>
#include <math.h>
// math 쓸 땐 -lm 옵션 붙여야함
// cd "/home/named/fastcamp/TIL/algorithm/boj/_1000/" && gcc 1011.c -o 1011 -lm && "/home/named/fastcamp/TIL/algorithm/boj/_1000/"1011

int main() {
    int t, x, y, r;
    scanf("%d", &t);
    for(int i=0; i<t; i++) {
        scanf("%d %d", &x, &y);
        r = floor(sqrt(y-x)+0.5);
        if(y-x <= pow(r, 2)) printf("%d\n", r*2-1);
        else printf("%d\n", r*2);
    }
    return 0;
}

File: algorithm/boj/_1000/1011.java
package TIL.algorithm.boj._1000;

import java.io.*;
import java.util.StringTokenizer;

class Main1011 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int t = Integer.parseInt(br.readLine());
        for(int i=0; i<t; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            double r = Math.floor(Math.sqrt(y-x)+0.5);
            if(y-x <= Math.pow(r, 2)) bw.write((int)r*2-1+"\n");
            else bw.write((int)r*2+"\n");
        }
        br.close();
        bw.flush();
        bw.close();
    }
}